test/system_SUITE.erl

```diff
@@ -0,0 +1,449 @@
+
+init_per_suite(Config) ->
+    rabbit_ct_helpers:log_environment(),
+    Config1 = rabbit_ct_helpers:set_config(Config, [
+        {rmq_nodename_suffix, ?MODULE}
+      ]),
+    Config2 = rabbit_ct_helpers:run_setup_steps(Config1,
+      rabbit_ct_broker_helpers:setup_steps() ++
+      rabbit_ct_client_helpers:setup_steps()),
+    Config2.
+
+end_per_suite(Config) ->
+    rabbit_ct_helpers:run_teardown_steps(Config,
+      rabbit_ct_client_helpers:teardown_steps() ++
+      rabbit_ct_broker_helpers:teardown_steps()).
+
+init_per_group(_, Config) ->
+    Config.
+
+end_per_group(_, Config) ->
+    Config.
+
+init_per_testcase(Testcase, Config) ->
+    ok = build_directory_tree(friendlies()),
+    rabbit_ct_helpers:testcase_started(Config, Testcase).
+
+end_per_testcase(Testcase, Config) ->
+    ok = force_delete_entire_directory(friendlies()),
+    rabbit_ct_helpers:testcase_finished(Config, Testcase).   
+
+
+%% -------------------------------------------------------------------
+%% Testsuite cases 
+%% -------------------------------------------------------------------
+
+library(_) ->
+     %% Given: Makefile.
+     {_Root, _Certificate, _Key} = ct_helper:make_certs(),
+     ok.
+
+invasive_SSL_option_change(Config) ->
+    ok = rabbit_ct_broker_helpers:rpc(Config, 0,
+          ?MODULE, invasive_SSL_option_change1, []).
+
+invasive_SSL_option_change1() ->
+    %% Given: Rabbit is started with the boot-steps in the
+    %% Trust-Store's OTP Application file.
+
+    %% When: we get Rabbit's SSL options.
+    {ok, Options} = application:get_env(rabbit, ssl_options),
+
+    %% Then: all necessary settings are correct.
+    verify_peer             = proplists:get_value(verify, Options),
+    true                    = proplists:get_value(fail_if_no_peer_cert, Options),
+    {Verifyfun, _UserState} = proplists:get_value(verify_fun, Options),
+
+    {module, rabbit_trust_store} = erlang:fun_info(Verifyfun, module),
+    {name,   whitelisted}        = erlang:fun_info(Verifyfun, name),
+    ok.
+
+validation_success_for_AMQP_client(Config) ->
+    ok = rabbit_ct_broker_helpers:rpc(Config, 0,
+           ?MODULE, validation_success_for_AMQP_client1, []). 
+
+validation_success_for_AMQP_client1() ->
+    AuthorityInfo = {Root, _AuthorityKey} = erl_make_certs:make_cert([{key, dsa}]),
+    {Certificate, Key} = chain(AuthorityInfo),
+    {Certificate2, Key2} = chain(AuthorityInfo),
+    Port = port(),
+    %% When: Rabbit accepts just this one authority's certificate
+    %% (i.e. these are options that'd be in the configuration
+    %% file).
+    ok = rabbit_networking:start_ssl_listener(Port, [{cacerts, [Root]},
+                                                     {cert, Certificate2},
+                                                     {key, Key2} | cfg()], 1),
+
+    %% Then: a client presenting a certifcate rooted at the same
+    %% authority connects successfully.
+    {ok, Con} = amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
```

Comment:
Please use `rabbit_ct_broker_helpers:get_config(Config, rmq_hostname)` instead of hardcoding an IP address.

Furthermore, this testcase would fail on an IPv6-only system.